From c77a96ccba91ca687fe0c22e3b2bb1141fbab421 Mon Sep 17 00:00:00 2001
From: Yunlian Jiang <yunlian@chromium.org>
Date: Tue, 28 Apr 2015 23:47:20 +0100
Subject: [PATCH] build: fix SINGLE_BINARY build when printf is a macro

* src/coreutils.c (usage): include coreutils.h outside
the printf call, because if it's a macro you will get the error:
embedding a #include directive within macro arguments is not supported
---
 src/coreutils.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/coreutils.c b/src/coreutils.c
index 9f798b8..7f86834 100644
--- a/src/coreutils.c
+++ b/src/coreutils.c
@@ -70,15 +70,17 @@ Execute the PROGRAM_NAME built-in program with the given PARAMETERS.\n\
       fputs (HELP_OPTION_DESCRIPTION, stdout);
       fputs (VERSION_OPTION_DESCRIPTION, stdout);
 
-      printf ("\n\
-Built-in programs:\n"
 #ifdef SINGLE_BINARY
-/* XXX: Ideally we#d like to present "install" here, not "ginstall".  */
+/* XXX: Ideally we'd like to present "install" here, not "ginstall".  */
+      char const *prog_name_list =
 # define SINGLE_BINARY_PROGRAM(prog_name_str, main_name) " " prog_name_str
 # include "coreutils.h"
 # undef SINGLE_BINARY_PROGRAM
+      ;
+      printf ("\n\
+Built-in programs:\n\
+%s\n", prog_name_list);
 #endif
-  "\n");
 
       printf (_("\
 \n\
-- 
2.3.6

